src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -295,21 +296,38 @@ impl<S: StateStore> StorageTable<S> {
         R: RangeBounds<B> + Send + Clone,
         B: AsRef<[u8]> + Send,
     {
-        // Vnodes that are set and should be accessed.
-        #[auto_enum(Iterator)]
-        let vnodes = match vnode_hint {
-            // If `vnode_hint` is set, we can only access this single vnode.
-            Some(vnode) => std::iter::once(vnode),
-            // Otherwise, we need to access all vnodes of this table.
-            None => self.vnodes.ones().map(VirtualNode::from_index),
+        let raw_key_ranges = if !ordered
+            && matches!(encoded_key_range.start_bound(), Unbounded)
+            && matches!(encoded_key_range.end_bound(), Unbounded)
+        {
+            // If the range is unbounded and order is not required, we can create a single iterator
+            // for each continuous vnode range.
+
+            // In this case, the `vnode_hint` must be default for singletons and `None` for
+            // distributed tables.
+            assert_eq!(vnode_hint.unwrap_or(DEFAULT_VNODE), DEFAULT_VNODE);
+
+            Either::Left(self.vnodes.high_ranges().map(|r| {
+                let start = Included(VirtualNode::from_index(*r.start()).to_be_bytes().to_vec());
+                let end = end_bound_of_prefix(&VirtualNode::from_index(*r.end()).to_be_bytes());
+                assert_matches!(end, Excluded(_) | Unbounded);
```

Comment:
Just to confirm when the last `vnode` = 255, `end` will be `Unbounded` ?

Reply:
Yes. The `end_bound` will be `Unbounded` if all bytes in the given prefix is '\xff'.